oras/provider.py

```diff
@@ -689,6 +707,10 @@ def push(self, *args, **kwargs) -> requests.Response:
         :type target: str
         :param refresh_headers: if true or None, headers are refreshed
         :type refresh_headers: bool
+        :param do_chunked: if true do chunked blob upload
```

Comment:
This would better be exposed through a separate `push_chunked` or similar function, if it's a  good idea to add it back.

Reply:
I don't understand this comment. It would require duplicating ~ 150 lines of code which  would be otherwise unchanged. Is that really what you want?

Reply:
Alternatively, if you like the separate function approach better than using an argument, I can rename existing 'push' to '_push' and create new, small 'push' and 'push_chunked'. Let me know how I should proceed.

Reply:
The initial design I had was fairly simple: https://github.com/oras-project/oras-py/blob/07ae3a1b46b245cb59dd390b50f3a900b07cd861/oras/provider.py#L147-L170 and based on size. I think we'd want to try a similar approach, with a boolean that defaults to False. For my comment above, I do think we want a clean separation of the two functionalities (akin to how it was done in the link above).